docs/migration/manual-migration.mdx

```diff
@@ -59,6 +58,37 @@ To cope with this change, you need to review all components using
   );
   ```
 
+## `withEntityNotFound` API change
+
+[`withEntityNotFound` API has changed](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/merge_requests/2354).
+In version 2, the parameter `NotFoundComponent` was optional (defaulted to
+`NotFound` component). However, this HOC is now deprecated and has been placed
+within `@front-commerce/compat` package, from which it is not possible to import
+files from the `theme`. Hence in v3, `withEntityNotFound`'s `NotFoundComponent`
+parameter is now mandatory.
```

Comment:
This is our internal implementation detail, I don't think it makes sense to expose all the reasons to readers.

**suggestion:** keep it simpler
```suggestion
`NotFound` component). This HOC is now deprecated and is now in the 
theme-agnostic `@front-commerce/compat` package.
If you still want to use `withEntityNotFound`, you must now provide a `NotFoundComponent`
component parameter.
```